odd-platform-ui/src/components/Search/Filters/FilterItem/MultipleFilterItem/MultipleFilterItemAutocomplete/MultipleFilterItemAutocomplete.tsx

```diff
@@ -0,0 +1,225 @@
+    option: FilterOption | null
+  ) => {
+    if (!option) return;
+    setSearchText(''); // Clear input on select
+    setFacets({
+      facetName,
+      facetOptionId: option.id,
+      facetOptionName: option.name,
+      facetOptionState: true,
+    });
+  };
+
+  const searchInputChange = React.useCallback(
+    (
+      _: React.ChangeEvent<unknown>,
+      query: string,
+      reason: AutocompleteInputChangeReason
+    ) => {
+      if (reason === 'input') {
+        setSearchText(query);
+      } else {
+        setSearchText(''); // Clear input on select
+      }
+    },
+    [setSearchText]
+  );
+
+  const getOptionLabel = React.useCallback(
+    (option: FilterOption) => option.name || '',
+    []
+  );
+
+  const getFilterOptions = React.useCallback(
+    (_, params) =>
+      filter(
+        searchText
+          ? facetOptions.filter(
+              option =>
+                option.name
+                  .toLocaleLowerCase()
+                  .indexOf(searchText.toLocaleLowerCase()) >= 0
+            )
+          : facetOptions,
+        params
+      ),
+    [searchText, facetOptions]
+  );
+
+  const [facetOptionsLoading, setFacetOptionsLoading] =
+    React.useState<boolean>(false);
+
+  const handleFacetSearch = React.useCallback(
+    useDebouncedCallback(() => {
+      setFacetOptionsLoading(true);
+      searchFacetOptions({
+        searchId,
+        facetType: facetName.toUpperCase() as MultipleFacetType,
+        page: 1,
+        size: 30,
+        query: searchText,
+      }).then(response => {
+        setFacetOptionsLoading(false);
+        setFacetOptions(response);
+      });
+    }, 500),
+    [
+      searchFacetOptions,
+      setFacetOptionsLoading,
+      setFacetOptions,
+      searchText,
+    ]
+  );
+
+  React.useEffect(() => {
+    if (!autocompleteOpen) return;
+    setFacetOptionsLoading(true);
+    handleFacetSearch();
+  }, [searchText, autocompleteOpen]);
+
+  const fillOptionMatches = (
+    props: HTMLAttributes<HTMLLIElement>,
+    option: FilterOption,
+    state: AutocompleteRenderOptionState
+  ) => {
+    const formattedOptionName =
+      facetName === 'types'
+        ? option.name.replaceAll('_', ' ')
+        : option.name;
+    if (!state.inputValue) {
+      return (
+        <li {...props}>
+          <Grid container justifyContent="space-between">
+            <span>{formattedOptionName}</span>
+            <FilterCount>{option.count}</FilterCount>
+          </Grid>
+        </li>
+      );
+    }
+
+    const highlightedText = (text: string, highlight: string) => {
+      const parts = text.split(new RegExp(`(${highlight})`, 'gi'));
+      return (
+        <span>
+          {parts.map((part, i) => (
+            <span
+              // eslint-disable-next-line react/no-array-index-key
+              key={i}
+              style={
+                part.toLowerCase() === highlight.toLowerCase()
+                  ? { backgroundColor: '#FFEECC' }
```

Comment:
(optional) create a styled component with background using palette colour 'orange10' instead of '#FFEECC'.